backend/inspirehep/editor/views.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from flask import Blueprint, jsonify, request
+from invenio_db import db
+from invenio_records.models import RecordMetadata
+from sqlalchemy_continuum import transaction_class, version_class
+
+from inspirehep.accounts.decorators import login_required_with_roles
+from inspirehep.accounts.roles import Roles
+from inspirehep.pidstore.api.base import PidStoreBase
+from inspirehep.records.api import InspireRecord
+
+MAX_UNIQUE_KEY_COUNT = 100
+
+blueprint = Blueprint("inspirehep_editor", __name__, url_prefix="/editor")
+
+
+@blueprint.route("/<endpoint>/<int:pid_value>/revisions/revert", methods=["PUT"])
+@login_required_with_roles([Roles.cataloger.value])
+def revert_to_revision(endpoint, pid_value):
+    """Revert given record to given revision"""
+    pid_type = PidStoreBase.get_pid_type_from_endpoint(endpoint)
+    record = InspireRecord.get_record_by_pid_value(pid_value, pid_type)
+    revision_id = request.json["revision_id"]
+    record.revert(revision_id)
+    db.session.commit()
+    return jsonify(success=True)
```

Comment:
we can gracefully handle exceptions, with https://github.com/inveniosoftware/invenio-rest/blob/master/invenio_rest/errors.py#L46 for example custom error and we can pass proper messages
```python
class EditorRevertToRevisionError(RESTException):
.....    
```

This can be done to all endpoints